### ISSUE TITLE
[ADAP-1067] [Regression] New default batch_id does not always conform to the required Dataproc batch ID pattern

### ISSUE THREAD

**Issue by dlubawy** [dbt-labs/dbt-bigquery#1050]
### Is this a regression in a recent version of dbt-bigquery?

- [X] I believe this is a regression in dbt-bigquery functionality
- [X] I have searched the existing issues, and I could not find an existing issue for this regression

### Current Behavior

BigQuery adapter will attempt to submit Dataproc serverless batch jobs for Python models using a combination of dbt project name, the node name of the model, and a timestamp as the batch ID. This will cause job failures for nodes with long names as Dataproc only supports IDs between 4 and 63 characters long.

### Expected/Previous Behavior

Default batch IDs were created using a randomly generated UUID between 4 and 63 character from the underlying Dataproc Python library. This meant dbt node names did not impact the building of Python models.

### Steps To Reproduce

1. Install v1.7 of `dbt-bigquery` and `dbt-core`
2. Create a Python model with a name longer than 63 characters
3. Build the dbt model

### Relevant log output

```shell
400 Batch ID 'model-xxx-xxxxxx-int-xxxx--python-xxx-xxxxxxxx-xxxxxxx-xxxx-xxxxx-xxxxxxxx1702065760' must conform to pattern '[a-z0-9][a-z0-9\-]{2,61}[a-z0-9]'
```

### Environment

```markdown
- OS: Darwin (macOS) 21.6.0
- Python: 3.9.18
- dbt-core (working version): 1.6.7
- dbt-bigquery (working version): 1.6.8
- dbt-core (regression version): 1.7.3
- dbt-bigquery (regression version): 1.7.2
```

**Comment by dlubawy** [comment]
Looks like I missed that #1020 would fix this issue and #1051 as well. This can probably be closed as a duplicate then since that PR is tracked in #1006.

**Comment by dbeatty10** [comment]
@dlubawy thanks for reporting the issues related to `batch_id` 🏆

Closing this one as a duplicate of #1006.
